src/libraries/System.Private.CoreLib/src/System/Threading/ThreadBlockingInfo.cs

```diff
@@ -185,8 +128,10 @@ public void Dispose()
 
         public enum ObjectKind // may be used by debuggers
         {
-            MonitorLock, // maps to DebugBlockingItemType::DebugBlock_MonitorCriticalSection in coreclr
-            MonitorWait, // maps to DebugBlockingItemType::DebugBlock_MonitorEvent in coreclr
+            [Obsolete("Represents native Monitor locks, which have been removed", error: true)]
+            MonitorLock, // maps to DebugBlockingItemType::Legacy_DebugBlock_MonitorCriticalSection in coreclr
+            [Obsolete("Represents native Monitor waits, which have been removed", error: true)]
+            MonitorWait, // maps to DebugBlockingItemType::Legacy_DebugBlock_MonitorEvent in coreclr
```

Comment:
```suggestion
            [Obsolete("Represents native Monitor locks, which have been removed", error: true)]
            MonitorLock,
            [Obsolete("Represents native Monitor waits, which have been removed", error: true)]
            MonitorWait,
```
This PR is deleting Legacy_DebugBlock_MonitorCriticalSection/Legacy_DebugBlock_MonitorEvent so there is no useful in referencing them from these comments.